<?xml version="1.0" ?>
<launch>
	<arg name = "use_rviz" default = "true" doc = "whether to use rviz"/>
	<rosparam command = "load" file = "$(find nusim)/config/basic_world.yaml"/>
	<rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />
	<arg name = "robot" default = "nusim"/>
	<arg name="odom_frame_id" default="odom"/>
	<include file = "$(find nuturtle_control)/launch/basic_remote.launch" if = "$(eval robot != 'nusim')">
		<arg name = "robot" value = "$(arg robot)"/>
	</include>

	<include file = "$(find nuturtle_control)/launch/start_robot.launch" if = "$(eval robot == 'nusim')">
		<arg name = "use_rviz" value = "false"/>
		<arg name = "cmd_src" default = "teleop"/>
		<param name="odom_frame_id" value ="$(arg odom_frame_id)"/>
	</include>
	<node pkg="nuslam" type="slam" name="slam"/>
	<node name="rviz" pkg="rviz" type="rviz" args="-d '$(find nuslam)/config/config.rviz'"/>
</launch>